src/cmd/apply.rs

```diff
@@ -154,14 +162,14 @@ pub fn run(argv: &[&str]) -> CliResult<()> {
 
         match &args.flag_new_column {
             Some(_) => {
-                record.push_field(cell.as_bytes());
+                record.push_field(&cell);
             }
             None => {
-                record = replace_column_value(&record, column_index, cell.as_bytes());
+                record = replace_column_value(&record, column_index, &cell);
             }
         }
 
-        wtr.write_byte_record(&record)?;
+        wtr.write_record(&record)?;
```

Comment:
This kind of change, FWIW, would be inappropriate in xsv because it would now prevent this command from being used with non-UTF-8 data, which is fairly common with messy CSV files.

Reply:
Yes I agree it would probably fit another tool which would require to work on correctly formatted CSV data, which is almost always my case. It was just convenient for me to have all of this in the same package (I add other tools for this before but none was close to xsv's speed, even though I am perfectly aware that I could reimplement the same logic in rust using your libraries in another tool altogether). Nonetheless a disclaimer in the docstring that this command will fail on non-UTF-8 data would probably be useful, on my end of course.